pandas/core/groupby/generic.py

```diff
@@ -1553,10 +1553,14 @@ def nunique(self, dropna: bool = True) -> DataFrame:
 
         return results
 
-    @doc(_shared_docs["idxmax"])
-    def idxmax(self, axis=0, skipna: bool = True):
+    @doc(
+        _shared_docs["idxmax"],
+        numeric_only_default="True for axis=0, False for axis=1",
+    )
+    def idxmax(self, axis=0, skipna: bool = True, numeric_only: bool | None = None):
         axis = DataFrame._get_axis_number(axis)
-        numeric_only = None if axis == 0 else False
+        if numeric_only is None:
+            numeric_only = None if axis == 0 else False
```

Comment:
Do you think this `None/False` behavior should be deprecated and changed in the future too to match the other ops?

Reply:
Yep! This PR adds numeric_only to the necessary ops, I have two follow ons that deprecate for frame and groupby ops respectively that are just about ready.